doc/cephfs/multifs.rst

```diff
@@ -0,0 +1,50 @@
+.. _cephfs-multifs:
+
+Multiple Ceph File Systems
+==========================
+
+
+Beginning with the Pacific release, multiple file system support is stable
+and ready to use. This functionality allows configuring separate file systems
+with full data separation on separate pools.
+
+Existing clusters must set a flag to enable multiple file systems::
+
+    ceph fs flag set enable_multiple true
+
+New Ceph clusters automatically set enable this.
+
+
+Creating a new Ceph File System
+-------------------------------
+
+The new ``volumes`` plugin interface (see: :doc:`/cephfs/fs-volumes`) automates
+most of the work of configuring a new file system. The "volume" concept is
+simply a new file system. This can be done via::
+
+    ceph fs volume create <fs_name>
+
+Ceph will create the new pools and automate the deployment of new MDS to
+support the new file system.
+
+
+Securing access
+---------------
+
+The ``fs authorize`` command allows configuring the client's access to a
+particular file system. See also in:ref:`fs-authorize-multifs`. The client will
+only have visibility of authorized file systems and the Monitors/MDS will
+reject access to clients without authorization.
+
+
+Other Notes
+-----------
+
+Multiple file systems do not share pools. This is particularly important for
+snapshots but also because no measures are in place to prevent duplicate
+inodes. The Ceph commands prevent this dangerous configuration.
+
+Each file system has its own set of MDS ranks. This can be useful for
```

Comment:
Note that my experience here has been mixed as MDS's are somewhat "promiscuous". I often find that after restarting the cluster that the MDS's get assigned to the different filesystems somewhat at random.

`mds: test:1/1 scratch:1/1 kbuild:1/1 {kbuild:0=test.ceph.kkaxpr=up:replay,scratch:0=scratch.ceph.gwkpaz=up:replay,test:0=kbuild.ceph.xpoutk=up:replay}`

So here, the kbuild and test filesystems swapped their MDS's. It's not clear to me that this is a bug, per-se, but it's misleading given the mds naming conventions.

Reply:
Oh, this may be a bug in the filesystem setup. It looks like the mds_join_fs value may be getting set on the wrong "who".

Reply:
https://docs.ceph.com/en/latest/cephfs/standby/#configuring-mds-file-system-affinity

We can link to the above doc. In your cluster, I wonder if it does eventually reach the right state? There can be a few failovers at the beginning.

Reply:
No, it never seems to reach the right state, but that may be due to lack of standby mds's. I'll just assume it's my issue for now.